lib/expression/v2/autodoc.ex

```diff
@@ -0,0 +1,266 @@
+defmodule Expression.V2.Autodoc do
```

Comment:
copied from V1 and updated for the new approach, mostly removing a lot of `eval!` and `eval_args!` calls.